addons/stock/tests/common.py

```diff
@@ -179,6 +223,84 @@ def setUpClass(cls):
     def url_extract_rec_id_and_model(self, url):
         # Extract model and record ID
         action_match = re.findall(r'action-([^/]+)', url)
-        model_name = self.env.ref(action_match[0]).res_model
+        model_name = self.env.ref(action_match[0]).sudo().res_model
         rec_id = re.findall(r'/(\d+)$', url)[0]
         return rec_id, model_name
+
+    @classmethod
+    def setup_logistics_user(cls):
+        cls.user = cls.env['res.users'].create({
+            'name': 'Test Stock User',
+            'login': 'stock',
+            'password': 'stockpass',
+            'groups_id': [
+                Command.link(cls.env.ref('stock.group_stock_manager').id),
+            ],
+        })
+        cls.user.email = 'stock@test.com'
+        # Shadow the current environment/cursor with one having the report user.
+        # This is mandatory to test access rights.
+        cls.env = cls.env(user=cls.user)
+        cls.cr = cls.env.cr
+        cls.assign_company_to_user(cls.user, cls.company)
```

Comment:
Personal opinion, but I really don't like modifying the environment and current user in the middle of the setup.
IMHO, setups shouldn't 'test' anything, if we want to test access rights, we should do so explicitly, and at least outside of the setup.

If you nevertheless decide to keep this approach, why don't you rely on the existing api of `setup_independent_user` and `get_default_groups` to create a generic test user, which will be used early in the chain of `setUpClass` because directly considered in the `BaseCommon` ?

Reply:
What i mean is that IMO `setUpClass` overrides in commons should not change the cursor/user/environment, at most they can imply some feature groups.  As you answered yourself, modifying the current user/environment implies that subsequent overrides/methods have to consider the possibility that the main user might not be the admin, and add sudos or additional rights here and there.

Creating some default users is a good idea (in your case, on stock user and one stock manager), but they should only be used manually in the setup of some test classes (in opposition to commons), or directly used when needed (either in a specific test, or with the dedicated `@users` decorator).

I mentioned `setup_independent_user` because IF we believe that modifying the setup environment and user is really what we want, I believe using the same API everywhere is the best.  There might be issues caused by that indeed (and that's a big reason why I don't like this approach as explained previously), but then those should be fixed to still have a coherent API.

I'm surprise about this cursor stuff, do you have a test in mind that requires that cursor change? I'd be curious to learn about it.  IMO, if some tests require a custom cursor, it should be moved in those specific tests or class setups instead of being part of such a core common).

Reply:
I've switched to `setup_independent_user` and `setup_independent_company` and added `.sudo()` in setup whenever necessary. It seems that there is no problem with the cursor anymore, so I'll remove that line. Could you please take another look?